packages/core/src/lib/use-keypress.mts

```diff
@@ -11,12 +11,15 @@ export function useKeypress(
   signal.current = userHandler;
 
   useEffect((rl) => {
+    let ignore = false;
```

Comment:
Went with this instead of the timeout approach. I think it's fair to ignore any events triggered after the cleanup are called.

Reply:
This is a fair approach. Do note though that since the `close` event is triggered before the `keypress` event, key combinations like `CTRL+C` won't be propagated to the `useKeypress` hook, so the behavior is slightly different from before.

I do think this behavior is the correct behavior, as those events are automatically handled by readline, so no custom logic on within the `useKeypress` callback should be necessary